lib/src/Base/Stat/CorrelationAnalysis.cxx

```diff
@@ -43,63 +88,34 @@ Point CorrelationAnalysis::PearsonCorrelation(const Sample & inputSample,
-                               const Sample & outputSample,
-                               const Bool normalize)
+/* Compute the squared Standard Regression Coefficients (SRC) between the input sample and the output sample */
+Point CorrelationAnalysis::computeSquaredSRC(const Bool normalize) const
 {
-  const UnsignedInteger dimension = inputSample.getDimension();
-  if (!(dimension >= 2)) throw InvalidDimensionException(HERE) << "Error: input sample must have dimension > 1, here dimension=" << dimension;
-  if (outputSample.getDimension() != 1) throw InvalidDimensionException(HERE) << "Error: output sample must be 1D";
-  if (inputSample.getSize() != outputSample.getSize()) throw InvalidArgumentException(HERE) << "Error: input and output samples must have the same size";
-  // If normalize, then sum is equal to 1
-  // and no information about noise
-  // Othewise we use standard formulation of src
-  // normalize false ==> SRC = SignedSRC * SignedSRC
-  if (!normalize)
-  {
-    // Case normalize false : coefficients are square of SignedSRC
-    Point src(SignedSRC(inputSample, outputSample));
-    for (UnsignedInteger i = 0; i < dimension; ++ i) src[i] = (src[i] * src[i]);
-    return src;
-  }
-
-  // Case where normalize is true
-  // Var(X+a) = Var(X); However for numerical stability, data are centered
-  LinearLeastSquares regressionAlgorithm(inputSample - inputSample.computeMean(), outputSample);
-  regressionAlgorithm.run();
-  // Linear coefficients
-  const Point linear(*regressionAlgorithm.getLinear().getImplementation());
-
-  // Compute the output variance from the regression coefficients
-  Point src(inputSample.computeVariance());
-  for (UnsignedInteger i = 0; i < dimension; ++ i) src[i] *= linear[i] * linear[i];
-  Scalar varOutput = 0.0;
-  for (UnsignedInteger i = 0; i < dimension; ++ i) varOutput += src[i];
-  if (varOutput > 0.0)
-    src /= varOutput;
-  else
-    throw InvalidArgumentException(HERE) << "No output variance";
+  Point src(computeSRC());
+  for (UnsignedInteger i = 0; i < src.getDimension(); ++ i) src[i] *= src[i];
+  // Case normalize true : coefficients are made to sum to 1
+  if (normalize) src /= src.norm1();
   return src;
 }
 
 /* Compute the Standard Regression Coefficients (SRC) between the input sample and the output sample */
-Point CorrelationAnalysis::SignedSRC(const Sample & inputSample,
-                                     const Sample & outputSample)
+Point CorrelationAnalysis::computeSRC() const
+{
+  return computeSRC(inputSample_, outputSample_);
+}
+
+// Compute the Standard Regression Coefficients (SRC) with arguments
+Point CorrelationAnalysis::computeSRC(const Sample & inputSample,
+                                      const Sample & outputSample) const
 {
   const UnsignedInteger dimension = inputSample.getDimension();
   if (!(dimension >= 2)) throw InvalidDimensionException(HERE) << "Error: input sample must have dimension > 1, here dimension=" << dimension;
```

Comment:
An SRC value could be computed between 2 1-dimensional samples, could it not?

Reply:
Actually, there is no problem at all. I don't know why why this check was made in the first place. Removed now.